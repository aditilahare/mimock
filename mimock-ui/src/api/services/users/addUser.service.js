import { post } from 'api/AxiosClient';
import Bcrypt from 'bcryptjs';

export const addNewUser = async ({ name, userName, password, userRole }) => {
	const salt = Bcrypt.genSaltSync(12);
	const hashedPassword = Bcrypt.hashSync(password, salt);

	return await post(
		'/admin/users',
		{
			name,
			userName,
			userRole,
			password: hashedPassword,
		},
		undefined,
		{ showFullPageLoader: true }
	).catch((err) => Promise.reject(err));
};
