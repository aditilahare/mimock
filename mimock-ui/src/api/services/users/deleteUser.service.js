import { remove } from 'api/AxiosClient';

export const deleteUser = async (userName) => {
	const url = '/admin/users/delete-user?userName=' + userName;

	return await remove(url)
		.then((res) => res.data)
		.catch((err) => Promise.reject(err));
};
