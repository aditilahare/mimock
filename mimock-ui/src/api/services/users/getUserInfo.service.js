import { get } from 'api/AxiosClient';

export const getUserInfo = async (userId) => {
	const url = `/admin/users/user-info?userId=${userId}`;

	return await get(url, { showFullPageLoader: true })
		.then((res) => res.data.userInfo)
		.catch((err) => Promise.reject(err));
};
