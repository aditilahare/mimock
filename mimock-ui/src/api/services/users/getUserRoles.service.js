import { get } from 'api/AxiosClient';

export const getUserRoles = async () => {
	return await get('/static-records/user-roles')
		.then((res) => res.data)
		.catch((err) => err);
};
