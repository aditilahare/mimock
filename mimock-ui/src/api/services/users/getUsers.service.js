import { get } from 'api/AxiosClient';
import { getUserDetails } from 'utils/jwtUtils';

export const getAllUsers = async () => {
	let userDetails;

	try {
		userDetails = getUserDetails();
	} catch (err) {
		return Promise.reject(err);
	}

	if (userDetails && userDetails.userRole === 'ROLE_ADMIN') {
		return await get('/admin/users', {
			showFullPageLoader: true,
		}).then((res) => {
			const { data } = res;
			return data.filter((user) => user.userName !== userDetails.userName);
		});
	}

	throw new Error('Unauthorized to access this resource');
};
