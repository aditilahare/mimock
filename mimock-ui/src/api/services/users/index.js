import { getAllUsers } from './getUsers.service';
import { deleteUser } from './deleteUser.service';

export { getAllUsers, deleteUser };
