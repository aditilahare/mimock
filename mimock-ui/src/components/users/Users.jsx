import React from 'react';
import UserManagement from './UserManagement';

export default function Users() {
	return <UserManagement />;
}
