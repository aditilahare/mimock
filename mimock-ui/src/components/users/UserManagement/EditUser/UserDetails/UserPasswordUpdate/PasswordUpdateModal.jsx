import React, { useState } from 'react';
import PropTypes from 'prop-types';
import { useRecoilState } from 'recoil';
import PropagateLoader from 'react-spinners/PropagateLoader';
import { ButtonVariants } from 'styles/Button';
import editUserDetailsAtom from 'atoms/editUserDetailsAtom';
import { updatePassword } from 'services/users';
import useNotification from 'hooks/useNotification';
import { notificationTypes } from 'constants/notificationConstants';
import {
	PasswordModalWrapper,
	Header,
	PasswordContainer,
	Label,
	PasswordInput,
	ButtonContainer,
	PasswordError,
	LoaderStyle,
	PasswordUpdateForm,
	CancelButton,
	UpdateButton,
} from './PasswordUpdateModal.style';

function PasswordUpdateModal({ userName }) {
	const [userInfo, setUserInfo] = useRecoilState(editUserDetailsAtom);
	const [updatingPassword, setUpdatingPassword] = useState(false);
	const [password, setPassword] = useState({
		newPassword: '',
		confirmPassword: '',
	});
	const [passwordError, setPasswordError] = useState({
		showError: false,
		errorMessage: '',
	});

	const newPasswordChange = (e) => {
		setPasswordError(false);
		setPassword({
			...password,
			newPassword: e.target.value,
		});
	};

	const confirmPasswordChange = (e) => {
		setPasswordError(false);
		setPassword({
			...password,
			confirmPassword: e.target.value,
		});
	};

	const passwordContainer = (label, testId, value, changeHandler) => {
		return (
			<PasswordContainer key={testId} data-testid={`${testId}-container`}>
				<Label>{label}</Label>
				<PasswordInput
					data-testid={`${testId}-input`}
					autoFocus={label === 'New Password' ? true : false}
					type='password'
					value={value}
					onChange={changeHandler}
					maxLength={128}
				/>
			</PasswordContainer>
		);
	};

	const closeModal = () => {
		setUserInfo({
			...userInfo,
			showPasswordUpdateModal: false,
		});
	};

	const isPasswordValid = () => {
		if (!password.newPassword || !password.confirmPassword) {
			setPasswordError({
				showError: true,
				errorMessage: 'Please enter password to continue',
			});
			return false;
		}

		if (password.newPassword !== password.confirmPassword) {
			setPasswordError({
				showError: true,
				errorMessage: 'Passwords do not match',
			});
			return false;
		}

		if (password.newPassword.length < 8) {
			setPasswordError({
				showError: true,
				errorMessage: 'Password must be at least 8 characters long',
			});
			return false;
		}

		if (password.newPassword.length > 128) {
			setPasswordError({
				showError: true,
				errorMessage: 'Password cannot be more than 128 characters',
			});
			return false;
		}

		return true;
	};

	const updatePasswordHandler = (e) => {
		e.preventDefault();
		if (!isPasswordValid()) {
			return;
		}
		setUpdatingPassword(true);

		updatePassword(userName, password.newPassword)
			.then(() => {
				setUpdatingPassword(false);
				useNotification({
					type: notificationTypes.NOTIFICATION_TYPE_SUCCESS,
					title: 'Success',
					message: 'Password updated successfully',
					animationIn: 'animate__slideInRight',
					animationOut: 'animate__slideOutRight',
				});
				closeModal();
			})
			.catch(() => {
				setUpdatingPassword(false);
				useNotification({
					type: notificationTypes.NOTIFICATION_TYPE_ERROR,
					title: 'Please try again',
					message: 'Failed to update password',
					animationIn: 'animate__slideInRight',
					animationOut: 'animate__slideOutRight',
				});
			});
	};

	return (
		<PasswordModalWrapper data-testid='password-update-modal'>
			<Header data-testid='password-update-header'>
				Updating password for {userName}
			</Header>
			<Choose>
				<When condition={updatingPassword}>
					<Header data-testid='password-update-loading-header'>
						Please wait...
					</Header>
					<PropagateLoader
						loading={updatingPassword}
						color='teal'
						size={30}
						css={LoaderStyle}
					/>
				</When>
				<Otherwise>
					<PasswordUpdateForm onSubmit={updatePasswordHandler}>
						{passwordContainer(
							'New Password',
							'new-password',
							password.newPassword,
							newPasswordChange
						)}
						{passwordContainer(
							'Confirm password',
							'confirm-password',
							password.confirmPassword,
							confirmPasswordChange
						)}
						<If condition={passwordError.showError}>
							<PasswordError data-testid='password-update-error'>
								{passwordError.errorMessage}
							</PasswordError>
						</If>
						<ButtonContainer>
							<UpdateButton
								type='submit'
								dataTestid='password-update-confirm-button'
								variant={ButtonVariants.GreenButton}
								label='Update'
								required={true}
								width='w-1/2'
							/>
							<CancelButton
								type='button'
								dataTestid='password-update-cancel-button'
								variant={ButtonVariants.RedButton}
								label='Cancel'
								width='w-1/2'
								required={true}
								onclickHandler={() => {
									closeModal();
								}}
							/>
						</ButtonContainer>
					</PasswordUpdateForm>
				</Otherwise>
			</Choose>
		</PasswordModalWrapper>
	);
}

PasswordUpdateModal.propTypes = {
	userName: PropTypes.string.isRequired,
};

export default PasswordUpdateModal;
