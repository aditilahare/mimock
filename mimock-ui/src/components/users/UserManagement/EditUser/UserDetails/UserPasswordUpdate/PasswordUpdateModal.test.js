import React from 'react';
import { act, render, fireEvent } from '@testing-library/react';
import { useRecoilState } from 'recoil';
import { updatePassword } from 'services/users';
import PasswordUpdateModal from './PasswordUpdateModal';

jest.setTimeout(8000);

let mockedRecoilFn = jest.fn();
jest.mock('recoil');

jest.mock('services/users');

const actualNotification = jest.requireActual('hooks/useNotification');
jest.mock('react-notifications-component', () => {
	const Store = {
		addNotification: jest.fn(),
	};
	return { Store };
});

describe('PasswordUpdateModal', () => {
	beforeEach(() => {
		useRecoilState.mockImplementation(() => {
			return [
				{
					userName: 'test1',
					name: 'Tester',
					isUserActive: false,
					userRole: 'MANAGER',
					passwordUpdatedAt: '2022-04-28 20:35:35',
					showPasswordUpdateModal: true,
				},
				mockedRecoilFn,
			];
		});
	});

	it('should render the PasswordUpdateModal', async () => {
		let tree;
		await act(async () => {
			tree = await render(<PasswordUpdateModal userName='test1' />);
		});

		const { getByTestId, queryByTestId, container } = tree;

		expect(getByTestId('password-update-modal')).toBeInTheDocument();
		expect(getByTestId('password-update-header')).toBeInTheDocument();
		expect(getByTestId('password-update-header')).toHaveTextContent(
			'Updating password for test1'
		);
		expect(getByTestId('new-password-container')).toBeInTheDocument();
		expect(getByTestId('confirm-password-container')).toBeInTheDocument();
		expect(getByTestId('password-update-cancel-button')).toBeInTheDocument();
		expect(getByTestId('password-update-confirm-button')).toBeInTheDocument();

		expect(
			queryByTestId('password-update-loading-header')
		).not.toBeInTheDocument();
		expect(queryByTestId('updating-password-loader')).not.toBeInTheDocument();
		expect(queryByTestId('password-update-error')).not.toBeInTheDocument();

		expect(container).toMatchSnapshot();
	});

	it('should show loader when updating password', async () => {
		const notificationSpy = jest.spyOn(actualNotification, 'default');
		updatePassword.mockImplementation(async () => {
			return await new Promise((resolve) => setTimeout(resolve, 3000));
		});

		let tree;
		await act(async () => {
			tree = await render(<PasswordUpdateModal userName='test1' />);
		});

		const { getByTestId, container } = tree;

		await act(async () => {
			await fireEvent.change(getByTestId('new-password-input'), {
				target: { value: 'password123' },
			});
			await fireEvent.change(getByTestId('confirm-password-input'), {
				target: { value: 'password123' },
			});
			await fireEvent.click(getByTestId('password-update-confirm-button'));
		});

		expect(getByTestId('password-update-loading-header')).toBeInTheDocument();
		expect(getByTestId('password-update-loading-header')).toHaveTextContent(
			'Please wait...'
		);
		expect(updatePassword).toHaveBeenCalledTimes(1);
		expect(updatePassword).toHaveBeenCalledWith('test1', 'password123');
		expect(notificationSpy).toHaveBeenCalledTimes(0);

		expect(container).toMatchSnapshot();
	});

	it('should show password length error when password is less than 8 characters long', async () => {
		const notificationSpy = jest.spyOn(actualNotification, 'default');
		updatePassword.mockResolvedValue({});

		let tree;
		await act(async () => {
			tree = await render(<PasswordUpdateModal userName='test1' />);
		});

		const { getByTestId, queryByTestId, container } = tree;

		await act(async () => {
			await fireEvent.change(getByTestId('new-password-input'), {
				target: { value: 'pass123' },
			});
			await fireEvent.change(getByTestId('confirm-password-input'), {
				target: { value: 'pass123' },
			});
			await fireEvent.click(getByTestId('password-update-confirm-button'));
		});

		expect(
			queryByTestId('password-update-loading-header')
		).not.toBeInTheDocument();

		expect(updatePassword).toHaveBeenCalledTimes(0);
		expect(notificationSpy).toHaveBeenCalledTimes(0);
		expect(getByTestId('password-update-error')).toBeInTheDocument();
		expect(getByTestId('password-update-error')).toHaveTextContent(
			'Password must be at least 8 characters long'
		);

		expect(container).toMatchSnapshot();
	});

	it('should show password length error when password is greater than 128 characters long', async () => {
		const notificationSpy = jest.spyOn(actualNotification, 'default');
		updatePassword.mockResolvedValue({});

		let tree;
		await act(async () => {
			tree = await render(<PasswordUpdateModal userName='test1' />);
		});

		const { getByTestId, queryByTestId, container } = tree;

		await act(async () => {
			await fireEvent.change(getByTestId('new-password-input'), {
				target: {
					value:
						'SzTlcfBHyuAeEfNCmcKaTzEZR0ZROkG4ttlQGNhUNRiUGH2K4tZ7iPowCDzWQOHVn9Tk0XVCLUZuxeSr21MTbWArSPLzsnFgUOCPChETEU4FMiQf0jOiKB7GRzh7GGQik',
				},
			});
			await fireEvent.change(getByTestId('confirm-password-input'), {
				target: {
					value:
						'SzTlcfBHyuAeEfNCmcKaTzEZR0ZROkG4ttlQGNhUNRiUGH2K4tZ7iPowCDzWQOHVn9Tk0XVCLUZuxeSr21MTbWArSPLzsnFgUOCPChETEU4FMiQf0jOiKB7GRzh7GGQik',
				},
			});
			await fireEvent.click(getByTestId('password-update-confirm-button'));
		});

		expect(
			queryByTestId('password-update-loading-header')
		).not.toBeInTheDocument();

		expect(updatePassword).toHaveBeenCalledTimes(0);
		expect(notificationSpy).toHaveBeenCalledTimes(0);
		expect(getByTestId('password-update-error')).toBeInTheDocument();
		expect(getByTestId('password-update-error')).toHaveTextContent(
			'Password cannot be more than 128 characters'
		);

		expect(container).toMatchSnapshot();
	});

	it('should show password validation error when new and confirmed passwords do not match', async () => {
		const notificationSpy = jest.spyOn(actualNotification, 'default');
		updatePassword.mockResolvedValue({});

		let tree;
		await act(async () => {
			tree = await render(<PasswordUpdateModal userName='test1' />);
		});

		const { getByTestId, queryByTestId, container } = tree;

		await act(async () => {
			await fireEvent.change(getByTestId('new-password-input'), {
				target: { value: 'password123' },
			});
			await fireEvent.change(getByTestId('confirm-password-input'), {
				target: { value: 'password12' },
			});
			await fireEvent.click(getByTestId('password-update-confirm-button'));
		});

		expect(
			queryByTestId('password-update-loading-header')
		).not.toBeInTheDocument();

		expect(updatePassword).toHaveBeenCalledTimes(0);
		expect(notificationSpy).toHaveBeenCalledTimes(0);
		expect(getByTestId('password-update-error')).toBeInTheDocument();
		expect(getByTestId('password-update-error')).toHaveTextContent(
			'Passwords do not match'
		);

		expect(container).toMatchSnapshot();
	});

	it('should close modal on clicking cancel button', async () => {
		const notificationSpy = jest.spyOn(actualNotification, 'default');
		updatePassword.mockResolvedValue({});

		let tree;
		await act(async () => {
			tree = await render(<PasswordUpdateModal userName='test1' />);
		});

		const { getByTestId, queryByTestId } = tree;

		await act(async () => {
			await fireEvent.change(getByTestId('new-password-input'), {
				target: { value: 'password123' },
			});
			await fireEvent.change(getByTestId('confirm-password-input'), {
				target: { value: 'password123' },
			});
			await fireEvent.click(getByTestId('password-update-cancel-button'));
		});

		expect(
			queryByTestId('password-update-loading-header')
		).not.toBeInTheDocument();

		expect(updatePassword).toHaveBeenCalledTimes(0);
		expect(notificationSpy).toHaveBeenCalledTimes(0);

		expect(mockedRecoilFn).toHaveBeenCalledTimes(1);
		expect(mockedRecoilFn).toHaveBeenCalledWith({
			userName: 'test1',
			name: 'Tester',
			isUserActive: false,
			userRole: 'MANAGER',
			passwordUpdatedAt: '2022-04-28 20:35:35',
			showPasswordUpdateModal: false,
		});
	});

	it('should update password and close modal on clicking update button', async () => {
		const notificationSpy = jest.spyOn(actualNotification, 'default');
		updatePassword.mockResolvedValue({});

		let tree;
		await act(async () => {
			tree = await render(<PasswordUpdateModal userName='test1' />);
		});

		const { getByTestId, queryByTestId, container } = tree;

		await act(async () => {
			await fireEvent.change(getByTestId('new-password-input'), {
				target: { value: 'password123' },
			});
			await fireEvent.change(getByTestId('confirm-password-input'), {
				target: { value: 'password123' },
			});
			await fireEvent.click(getByTestId('password-update-confirm-button'));
		});

		expect(
			queryByTestId('password-update-loading-header')
		).not.toBeInTheDocument();

		expect(updatePassword).toHaveBeenCalledTimes(1);
		expect(updatePassword).toHaveBeenCalledWith('test1', 'password123');
		expect(notificationSpy).toHaveBeenCalledTimes(1);
		expect(notificationSpy).toHaveBeenCalledWith({
			type: 'success',
			title: 'Success',
			message: 'Password updated successfully',
			animationIn: 'animate__slideInRight',
			animationOut: 'animate__slideOutRight',
		});
		expect(mockedRecoilFn).toHaveBeenCalledTimes(1);
		expect(mockedRecoilFn).toHaveBeenCalledWith({
			userName: 'test1',
			name: 'Tester',
			isUserActive: false,
			userRole: 'MANAGER',
			passwordUpdatedAt: '2022-04-28 20:35:35',
			showPasswordUpdateModal: false,
		});

		expect(container).toMatchSnapshot();
	});

	it('should show error notification when password update api call fails', async () => {
		const notificationSpy = jest.spyOn(actualNotification, 'default');
		updatePassword.mockRejectedValue({});

		let tree;
		await act(async () => {
			tree = await render(<PasswordUpdateModal userName='test1' />);
		});

		const { getByTestId, queryByTestId, container } = tree;

		await act(async () => {
			await fireEvent.change(getByTestId('new-password-input'), {
				target: { value: 'password123' },
			});
			await fireEvent.change(getByTestId('confirm-password-input'), {
				target: { value: 'password123' },
			});
			await fireEvent.click(getByTestId('password-update-confirm-button'));
		});

		expect(
			queryByTestId('password-update-loading-header')
		).not.toBeInTheDocument();

		expect(updatePassword).toHaveBeenCalledTimes(1);
		expect(updatePassword).toHaveBeenCalledWith('test1', 'password123');
		expect(notificationSpy).toHaveBeenCalledTimes(1);
		expect(notificationSpy).toHaveBeenCalledWith({
			type: 'danger',
			title: 'Please try again',
			message: 'Failed to update password',
			animationIn: 'animate__slideInRight',
			animationOut: 'animate__slideOutRight',
		});
		expect(mockedRecoilFn).toHaveBeenCalledTimes(0);

		expect(container).toMatchSnapshot();
	});
});
