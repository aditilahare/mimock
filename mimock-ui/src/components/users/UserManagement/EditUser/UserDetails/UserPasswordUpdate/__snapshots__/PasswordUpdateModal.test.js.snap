// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PasswordUpdateModal should render the PasswordUpdateModal 1`] = `
<div>
  <div
    class="xl:w-3/4 md:w-11/12 bg-white mx-auto my-4 p-3 shadow-sm border border-gray-150 rounded-md"
    data-testid="password-update-modal"
  >
    <div
      class="font-sans subpixel-antialiased text-base text-center text-gray-400 mt-2 mb-10"
      data-testid="password-update-header"
    >
      Updating password for 
      test1
    </div>
    <form
      class=""
    >
      <div
        class="flex xl:w-1/2 md:w-2/4 mx-auto my-8 items-center align-middle justify-between"
        data-testid="new-password-container"
      >
        <label
          class="font-sans subpixel-antialiased text-gray-700 font-bold"
        >
          New Password
        </label>
        <div
          class="block h-12 min-h-12 max-h-12"
        >
          <input
            class="block px-3 py-1.5 text-base font-normal text-gray-700 bg-zinc-50 bg-clip-padding border border-solid rounded mx-4 focus:outline-blue-200 border-gray-300"
            data-testid="new-password-input"
            name="password"
            placeholder="Enter password"
            type="password"
          />
        </div>
      </div>
      <div
        class="flex xl:w-1/2 md:w-2/4 mx-auto my-8 items-center align-middle justify-between"
        data-testid="confirm-password-container"
      >
        <label
          class="font-sans subpixel-antialiased text-gray-700 font-bold"
        >
          Confirm password
        </label>
        <div
          class="block h-12 min-h-12 max-h-12"
        >
          <input
            class="block px-3 py-1.5 text-base font-normal text-gray-700 bg-zinc-50 bg-clip-padding border border-solid rounded mx-4 focus:outline-blue-200 border-gray-300"
            data-testid="confirm-password-input"
            name="confirmPassword"
            placeholder="Confirm password"
            type="password"
          />
        </div>
      </div>
      <div
        class="flex mx-auto justify-evenly my-10 w-1/2 gap-6 flex-row-reverse"
      >
        <button
          class="px-9 py-2 shadow-md rounded-md text-center text-base font-semibold cursor-pointer hover:brightness-105 outline-none text-lime-700 dark:text-white w-1/2 bg-green-300 dark:bg-green-400"
          data-testid="password-update-confirm-button"
        >
          Update
        </button>
        <button
          class="px-9 py-2 shadow-md rounded-md text-center text-base font-semibold cursor-pointer hover:brightness-105 outline-none text-red-800 dark:text-white w-1/2 bg-red-300 dark:bg-red-500"
          data-testid="password-update-cancel-button"
        >
          Cancel
        </button>
      </div>
    </form>
  </div>
</div>
`;

exports[`PasswordUpdateModal should show error notification when password update api call fails 1`] = `
<div>
  <div
    class="xl:w-3/4 md:w-11/12 bg-white mx-auto my-4 p-3 shadow-sm border border-gray-150 rounded-md"
    data-testid="password-update-modal"
  >
    <div
      class="font-sans subpixel-antialiased text-base text-center text-gray-400 mt-2 mb-10"
      data-testid="password-update-header"
    >
      Updating password for 
      test1
    </div>
    <form
      class=""
    >
      <div
        class="flex xl:w-1/2 md:w-2/4 mx-auto my-8 items-center align-middle justify-between"
        data-testid="new-password-container"
      >
        <label
          class="font-sans subpixel-antialiased text-gray-700 font-bold"
        >
          New Password
        </label>
        <div
          class="block h-12 min-h-12 max-h-12"
        >
          <input
            class="block px-3 py-1.5 text-base font-normal text-gray-700 bg-zinc-50 bg-clip-padding border border-solid rounded mx-4 focus:outline-blue-200 border-gray-300"
            data-testid="new-password-input"
            name="password"
            placeholder="Enter password"
            type="password"
          />
        </div>
      </div>
      <div
        class="flex xl:w-1/2 md:w-2/4 mx-auto my-8 items-center align-middle justify-between"
        data-testid="confirm-password-container"
      >
        <label
          class="font-sans subpixel-antialiased text-gray-700 font-bold"
        >
          Confirm password
        </label>
        <div
          class="block h-12 min-h-12 max-h-12"
        >
          <input
            class="block px-3 py-1.5 text-base font-normal text-gray-700 bg-zinc-50 bg-clip-padding border border-solid rounded mx-4 focus:outline-blue-200 border-gray-300"
            data-testid="confirm-password-input"
            name="confirmPassword"
            placeholder="Confirm password"
            type="password"
          />
        </div>
      </div>
      <div
        class="flex mx-auto justify-evenly my-10 w-1/2 gap-6 flex-row-reverse"
      >
        <button
          class="px-9 py-2 shadow-md rounded-md text-center text-base font-semibold cursor-pointer hover:brightness-105 outline-none text-lime-700 dark:text-white w-1/2 bg-green-300 dark:bg-green-400"
          data-testid="password-update-confirm-button"
        >
          Update
        </button>
        <button
          class="px-9 py-2 shadow-md rounded-md text-center text-base font-semibold cursor-pointer hover:brightness-105 outline-none text-red-800 dark:text-white w-1/2 bg-red-300 dark:bg-red-500"
          data-testid="password-update-cancel-button"
        >
          Cancel
        </button>
      </div>
    </form>
  </div>
</div>
`;

exports[`PasswordUpdateModal should show error passwords are empty 1`] = `
<div>
  <div
    class="xl:w-3/4 md:w-11/12 bg-white mx-auto my-4 p-3 shadow-sm border border-gray-150 rounded-md"
    data-testid="password-update-modal"
  >
    <div
      class="font-sans subpixel-antialiased text-base text-center text-gray-400 mt-2 mb-10"
      data-testid="password-update-header"
    >
      Updating password for 
      test1
    </div>
    <form
      class=""
    >
      <div
        class="flex xl:w-1/2 md:w-2/4 mx-auto my-8 items-center align-middle justify-between"
        data-testid="new-password-container"
      >
        <label
          class="font-sans subpixel-antialiased text-gray-700 font-bold"
        >
          New Password
        </label>
        <div
          class="block h-12 min-h-12 max-h-12"
        >
          <input
            class="block px-3 py-1.5 text-base font-normal text-gray-700 bg-zinc-50 bg-clip-padding border border-solid rounded mx-4 focus:outline-blue-200 border-2 border-rose-300"
            data-testid="new-password-input"
            name="password"
            placeholder="Enter password"
            type="password"
          />
          <div
            class="text-red-500 text-xs font-sans subpixel-antialiased mx-4 w-40 absolute"
            data-testid="input-error-password"
          >
            Password is required
          </div>
        </div>
      </div>
      <div
        class="flex xl:w-1/2 md:w-2/4 mx-auto my-8 items-center align-middle justify-between"
        data-testid="confirm-password-container"
      >
        <label
          class="font-sans subpixel-antialiased text-gray-700 font-bold"
        >
          Confirm password
        </label>
        <div
          class="block h-12 min-h-12 max-h-12"
        >
          <input
            class="block px-3 py-1.5 text-base font-normal text-gray-700 bg-zinc-50 bg-clip-padding border border-solid rounded mx-4 focus:outline-blue-200 border-2 border-rose-300"
            data-testid="confirm-password-input"
            name="confirmPassword"
            placeholder="Confirm password"
            type="password"
          />
          <div
            class="text-red-500 text-xs font-sans subpixel-antialiased mx-4 w-40 absolute"
            data-testid="input-error-confirmPassword"
          >
            The passwords do not match
          </div>
        </div>
      </div>
      <div
        class="flex mx-auto justify-evenly my-10 w-1/2 gap-6 flex-row-reverse"
      >
        <button
          class="px-9 py-2 shadow-md rounded-md text-center text-base font-semibold cursor-pointer hover:brightness-105 outline-none text-lime-700 dark:text-white w-1/2 bg-green-300 dark:bg-green-400"
          data-testid="password-update-confirm-button"
        >
          Update
        </button>
        <button
          class="px-9 py-2 shadow-md rounded-md text-center text-base font-semibold cursor-pointer hover:brightness-105 outline-none text-red-800 dark:text-white w-1/2 bg-red-300 dark:bg-red-500"
          data-testid="password-update-cancel-button"
        >
          Cancel
        </button>
      </div>
    </form>
  </div>
</div>
`;

exports[`PasswordUpdateModal should show loader when updating password 1`] = `
<div>
  <div
    class="xl:w-3/4 md:w-11/12 bg-white mx-auto my-4 p-3 shadow-sm border border-gray-150 rounded-md"
    data-testid="password-update-modal"
  >
    <div
      class="font-sans subpixel-antialiased text-base text-center text-gray-400 mt-2 mb-10"
      data-testid="password-update-header"
    >
      Updating password for 
      test1
    </div>
    <div
      class="font-sans subpixel-antialiased text-base text-center text-gray-400 mt-2 mb-10"
      data-testid="password-update-loading-header"
    >
      Please wait...
    </div>
    <span
      class="css-1uyxhex-LoaderStyle"
    >
      <span
        class="css-m7eth1"
      />
      <span
        class="css-16zptrm"
      />
      <span
        class="css-1ma23m2"
      />
      <span
        class="css-ilqonc"
      />
      <span
        class="css-gmg1hm"
      />
      <span
        class="css-1sh5wjb"
      />
    </span>
  </div>
</div>
`;

exports[`PasswordUpdateModal should update password and close modal on clicking update button 1`] = `
<div>
  <div
    class="xl:w-3/4 md:w-11/12 bg-white mx-auto my-4 p-3 shadow-sm border border-gray-150 rounded-md"
    data-testid="password-update-modal"
  >
    <div
      class="font-sans subpixel-antialiased text-base text-center text-gray-400 mt-2 mb-10"
      data-testid="password-update-header"
    >
      Updating password for 
      test1
    </div>
    <form
      class=""
    >
      <div
        class="flex xl:w-1/2 md:w-2/4 mx-auto my-8 items-center align-middle justify-between"
        data-testid="new-password-container"
      >
        <label
          class="font-sans subpixel-antialiased text-gray-700 font-bold"
        >
          New Password
        </label>
        <div
          class="block h-12 min-h-12 max-h-12"
        >
          <input
            class="block px-3 py-1.5 text-base font-normal text-gray-700 bg-zinc-50 bg-clip-padding border border-solid rounded mx-4 focus:outline-blue-200 border-gray-300"
            data-testid="new-password-input"
            name="password"
            placeholder="Enter password"
            type="password"
          />
        </div>
      </div>
      <div
        class="flex xl:w-1/2 md:w-2/4 mx-auto my-8 items-center align-middle justify-between"
        data-testid="confirm-password-container"
      >
        <label
          class="font-sans subpixel-antialiased text-gray-700 font-bold"
        >
          Confirm password
        </label>
        <div
          class="block h-12 min-h-12 max-h-12"
        >
          <input
            class="block px-3 py-1.5 text-base font-normal text-gray-700 bg-zinc-50 bg-clip-padding border border-solid rounded mx-4 focus:outline-blue-200 border-gray-300"
            data-testid="confirm-password-input"
            name="confirmPassword"
            placeholder="Confirm password"
            type="password"
          />
        </div>
      </div>
      <div
        class="flex mx-auto justify-evenly my-10 w-1/2 gap-6 flex-row-reverse"
      >
        <button
          class="px-9 py-2 shadow-md rounded-md text-center text-base font-semibold cursor-pointer hover:brightness-105 outline-none text-lime-700 dark:text-white w-1/2 bg-green-300 dark:bg-green-400"
          data-testid="password-update-confirm-button"
        >
          Update
        </button>
        <button
          class="px-9 py-2 shadow-md rounded-md text-center text-base font-semibold cursor-pointer hover:brightness-105 outline-none text-red-800 dark:text-white w-1/2 bg-red-300 dark:bg-red-500"
          data-testid="password-update-cancel-button"
        >
          Cancel
        </button>
      </div>
    </form>
  </div>
</div>
`;
